design-proposals/orch-cli.md

```diff
@@ -0,0 +1,229 @@
+# Design Proposal: Orchestrator Command Line Interface
+
+Author(s): Scott Baker
+
+Last updated: 2025-04-29
+
+## Abstract
+
+This ADR describes the design of a command line interface (CLI) for the
+Orchestrator. The CLI is a cross-component tool that may be used from a Linux, Windows, or
+Mac environment, such as from the administrator's laptop. The CLI may also may be leveraged
+as a building block in other environments such as CI/CD or testing.
+
+The CLI described here uses the `Catalog CLI` as a strating point and inspiration. However, this
+ADR assumes that syntax and semantics of the CLI may diverge from the existing `Catalog CLI` and
+this ADR is not constrained by that existing implementation. As such, examples given in this
+ADR may not necessarily work with the existing catalog CLI.
+
+## Proposal
+
+The CLI implements a set of verbs, a set of nouns, and a set of common options that may
+be used to interact with the orchestrator.
+
+The CLI follows a `verb - noun - subject` pattern. For example,
+
+- `cli get application nginx`. The verb is `get`. The noun is `application` and the subject
+  is `nginx`. This particular CLI command would return the application called nginx from the
+  Orchestrator and print them to the console in a human-readable format.
+
+### Syntax
+
+This section describes the syntax of the CLI.
+
+#### Verbs
+
+Note that some verbs may have synonyms. For example, `set` and `update` are synonyms for the same
+verb.
+
+##### Session Management
+
+- `login <user-id>` ... log in to the Orchestrator. The CLI shall prompt for a password and retrieve
+  a refresh token. The refresh token is cached locally until it is expired or a `logout` verb is used.
+
+- `logout` ... log out of the Orchestrator. Any active login state, including the refresh token, is
+  discarded.
+
+##### CRUD Operations
+
+- `list` ... lists objects on the Orchestrator. The `list` verb produces a list of all objects
```

Comment:
What are objects in this context: components/services deployed on the orchestrator or clusters/applications/ens managed by the orchestrator?

Reply:
In my opinion, clusters/applications/ens managed by the orchestrator. @se-chris-thach is working on a proposal for an installer CLI (potentially to be combined here) and maybe it would make sense to also have operations on the components, to start/stop/get-status on them either in that CLI, or in a combined CLI.